### PR TITLE
Update testing documentation

### LINKED ISSUE #114: Document configuration for AGP to unit test Molecule
I'm trying out Molecule with AAC ViewModel. Here is how the VM code is written exposing stateFlow stream using `launchMolecule()`

Molecule version used : `0.5.0-SNAPSHOT`

```kt
@HiltViewModel
class PhotosListViewModel @Inject constructor(
    private val unsplashRepository: UnsplashRepository,
    @MoleculeScope private val scope: CoroutineScope,
    @CompositionClock private val clock: RecompositionClock,
) : ViewModel() {

    private val events = Channel<Event>()
    val stateFlow = scope.launchMolecule(clock = clock) {
        present(events.receiveAsFlow())
    }

    init {
        processEvent(InitialPageEvent)
    }

    @Composable
    fun present(events: Flow<Event>): PhotosListUIState {
        var isLoading by remember { mutableStateOf(false) }
        var error by remember { mutableStateOf<String?>(null) }
        val images = remember { mutableStateListOf<UnsplashImage>() }

        LaunchedEffect(Unit) {
            events.collect { event ->
                when (event) {
                    InitialPageEvent -> {
                        isLoading = true
                        error = null
                        when (val result =
                            unsplashRepository.getPhotos(page = 1, perPage = ITEM_PER_PAGE)) {
                            is Error -> {
                                isLoading = false
                                error = result.message
                            }
                            is Success -> {
                                isLoading = false
                                images.addAll(result.data.images)
                            }
                        }

                    }
                }
            }
        }
        return PhotosListUIState(
            isLoading = isLoading,
            error = error,
            images = images,
        )
    }

    fun processEvent(event: Event) {
        scope.launch {
            events.send(event)
        }
    }
}

// UI State
data class PhotosListUIState(
    val isLoading: Boolean = false,
    val error: String? = null,
    val images: List<UnsplashImage> = emptyList(),
)

// Events
sealed interface Event
object InitialPageEvent : Event
```

App works fine on Android devices without any issue when VM is provided with correct coroutineScope and frameClock. However, when I write Unit test for VM it requires `android.os.Trace` to be mocked. Added both unit test and it's error log below.

VM's Unit Test
```kt
import app.cash.molecule.RecompositionClock
import app.cash.turbine.testIn
import dev.punitd.unplashapp.data.fake.FakeUnsplashRepository
import dev.punitd.unplashapp.model.images
import dev.punitd.unplashapp.model.pageLinks
import dev.punitd.unplashapp.util.CoroutineRule
import kotlinx.coroutines.ExperimentalCoroutinesApi
import kotlinx.coroutines.test.runTest
import org.junit.Assert.assertEquals
import org.junit.Rule
import org.junit.Test


@OptIn(ExperimentalCoroutinesApi::class)
class PhotosListViewModelTest {

    @get:Rule
    val coroutineRule = CoroutineRule()

    @Test
    fun successStateIfApiSucceeds() = runTest {
        val viewModel = PhotosListViewModel(
            unsplashRepository = FakeUnsplashRepository(coroutineRule.testDispatcher),
            scope = this, // TestScope
            clock = RecompositionClock.Immediate,
        )

        // InitialPageEvent is send inside VM's init{} block by default
        // That's why we're not sending any events here.

        val turbine = viewModel.stateFlow.testIn(this)
        assertEquals(PhotosListUIState(isLoading = false), turbine.awaitItem())
        assertEquals(PhotosListUIState(isLoading = true), turbine.awaitItem())
        assertEquals(
            PhotosListUIState(
                isLoading = false,
                error = null,
                images = images,
                pageLinks = pageLinks
            ),
            turbine.awaitItem()
        )
        turbine.cancel()
    }
}
``` 

Test Error log 
```
Method beginSection in android.os.Trace not mocked. See http://g.co/androidstudio/not-mocked for details.
java.lang.RuntimeException: Method beginSection in android.os.Trace not mocked. See http://g.co/androidstudio/not-mocked for details.
	at android.os.Trace.beginSection(Trace.java)
	at androidx.compose.runtime.Trace.beginSection(ActualAndroid.android.kt:30)
	at androidx.compose.runtime.ComposerImpl.doCompose(Composer.kt:4357)
	at androidx.compose.runtime.ComposerImpl.composeContent$runtime_release(Composer.kt:3119)
	at androidx.compose.runtime.CompositionImpl.composeContent(Composition.kt:584)
	at androidx.compose.runtime.Recomposer.composeInitial$runtime_release(Recomposer.kt:811)
	at androidx.compose.runtime.CompositionImpl.setContent(Composition.kt:519)
	at app.cash.molecule.MoleculeKt.launchMolecule(molecule.kt:163)
	at app.cash.molecule.MoleculeKt.launchMolecule(molecule.kt:108)
	at dev.punitd.unplashapp.screen.photos.PhotosListViewModel.<init>(PhotosListViewModel.kt:32)
```
I'm sure i'm doing something stupid here because unit tests written in `sample` app of this repo works just fine without any such error.

Questions: 
- Is there something I need to mock which i'm missing here? 
- Are Coroutinescope and FrameClock passed to VM in test incorrect? 

Repo to reproduce it : https://github.com/punitda/Tinysplash

Thanks!

### PR DESCRIPTION
**Pull request by punitda**
Changes : 
- Added AGP config instruction for unit testing molecules in Android
- Updated testing section to remove references to `molecule-testing` artifact.

Fixes #114

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -225,38 +225,35 @@ So all Molecule APIs require you to specify your preferred clock behavior:
...
+      assertEquals(2, awaitItem())
+      cancel()
+    }
 }
 ```
 
-Unlike `moleculeFlow(clock = Immediate)`, `testMolecule` simulates clock ticks:
-Calling `awaitItem` yields to running coroutines and "ticks" the Molecule's clock until a change to snapshot state occurs, triggering a fresh recomposition and a new item.
-If snapshot state is never changed, no recomposition will ever occur and `awaitItem` will fail.
+> Note : If you're earlier using `molecule-testing` dependency, please switch to testing your molecule using `moleculeFlow` with `ImmediateClock` and [Turbine](https://github.com/cashapp/turbine/).
```
Comment (oldergod) [inline README.md]:
```suggestion
```

File: README.md [hunk 2]
```diff
@@ -225,38 +225,35 @@ So all Molecule APIs require you to specify your preferred clock behavior:
...
-If snapshot state is never changed, no recomposition will ever occur and `awaitItem` will fail.
+> Note : If you're earlier using `molecule-testing` dependency, please switch to testing your molecule using `moleculeFlow` with `ImmediateClock` and [Turbine](https://github.com/cashapp/turbine/).  
 
-```kotlin
-@Test fun counting() {
-  testMolecule({ Counter(1, 3) }) {
-    assertEquals(1, awaitItem())
-    assertEquals(2, awaitItem())
-    assertEquals(3, awaitItem())
+If you're unit testing a molecule in Android, please set below in your project's AGP config.
```
Comment (oldergod) [inline README.md]:
```suggestion
If you use Molecule and reference Android APIs in your unit tests, you might need to set the following Android Gradle plugin option:
```

**Review by oldergod — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (JakeWharton) [reply README.md]: The Compose runtime is the library that calls into Android APIs. So all Molecule users in an Android module will need to do this.

File: README.md [hunk 3]
```diff
@@ -225,38 +225,35 @@ So all Molecule APIs require you to specify your preferred clock behavior:
...
-  // or androidTestImplementation…
+@Test fun counter() = runTest {
+  moleculeFlow(RecompositionClock.Immediate) {
+    Counter()
+  }.test {
+      assertEquals(0, awaitItem())
+      assertEquals(1, awaitItem())
+      assertEquals(2, awaitItem())
+      cancel()
+    }
```
Comment (JakeWharton) [inline README.md]:
```suggestion
  }.test {
    assertEquals(0, awaitItem())
    assertEquals(1, awaitItem())
    assertEquals(2, awaitItem())
    cancel()
  }
```

File: README.md (hunk 2 above)
Comment (JakeWharton) [inline README.md]:
```suggestion
If you're unit testing Molecule on the JVM in an Android module, please set below in your project's AGP config.
```

**Review by JakeWharton — APPROVED** [review]

**Pull request merged**